src/PAModel/Serializers/SourceSerializer.cs

```diff
@@ -375,9 +375,18 @@ private static void LoadSourceFiles(CanvasDocument app, DirectoryReader director
             }
 
             // When loading TestSuites sharded files, add them within the top parent AppTest control (i.e. Test_7F478737223C4B69)
```

Comment:
I'm wondering whether this code belongs here, or in the CanvasDocument.ApplyBefore/AfterLoad functions

Reply:
Any thoughts?

Reply:
I am not sure if this can be moved there, since its happening at the time of reading the source files. We read the source files and we add controls to add it to the app object. Once we have the app object then we perform ApplyBefore/AfterLoad functions in the SaveAsMsApp method.

Reply:
Fair enough.